ddtrace/contrib/pymongo/client.py

```diff
@@ -114,35 +114,29 @@ def _datadog_trace_operation(self, operation):
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
         return span
 
-    # Pymongo >= 3.9
-    def run_operation_with_response(self, sock_info, operation, *args, **kwargs):
-        span = self._datadog_trace_operation(operation)
-        if not span:
-            return self.__wrapped__.run_operation_with_response(sock_info, operation, *args, **kwargs)
+    # Pymongo >= 3.12
+    def run_operation(self, sock_info, operation, *args, **kwargs):
+        with self._datadog_trace_operation(operation) as span:
+            result = self.__wrapped__.run_operation(sock_info, operation, *args, **kwargs)
+            if result and result.address:
+                set_address_tags(span, result.address)
+            return result
 
-        try:
+    # Pymongo >= 3.9, <3.12
+    def run_operation_with_response(self, sock_info, operation, *args, **kwargs):
```

Comment:
should we have these within if statements?

what happens if we add `run_operation_with_response` to version >=3.12?

Reply:
We're just adding functions that don't get used.

I suppose the worst that could happen is a user is using this API and doesn't upgrade their code. In that case they'll see the wrapper function in the stack trace as `result = self.__wrapped__.run_operation_with_response(sock_info, operation, *args, **kwargs)` will raise.

I'll see if it's easy enough to version it though :+1:

Reply:
Just to clarify, `run_operation` and `run_operation_with_response` are the same API, just renamed for different versions of Pymongo (>=3.12 and >=3.9 respectively), right? I.e. `run_operation_with_response` won't be called for versions >=3.12?

Reply:
Correct: https://github.com/mongodb/mongo-python-driver/commit/c8f32a7a376e027cbe30dcb8df9bf05778e61248

Reply:
Question: How can we use `with self._datadog_trace_operation(operation) as span:`  when the function isn't a context manager?

Reply:
@stradigi-David-Malvar it's the return value of the function that has to be a context manager. `_datadog_trace_operation` returns a span which is a context manager.